src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerCoordinator.swift

```diff
@@ -135,34 +144,78 @@ class QRScannerCoordinator {
 	}
 
 	private func showQRCodeResult(qrCodeResult: QRCodeResult) {
-		parentViewController?.dismiss(animated: true, completion: { [weak self] in
-			switch qrCodeResult {
-			case let .coronaTest(testRegistrationInformation):
-				self?.showScannedTestResult(testRegistrationInformation)
-			case let .certificate(certificateResult):
-				self?.showScannedHealthCertificate(certificateResult)
-			case let .traceLocation(traceLocation):
-				self?.showScannedCheckin(traceLocation)
-			}
-		})
+		switch qrCodeResult {
+		case let .coronaTest(testRegistrationInformation):
+			showScannedTestResult(testRegistrationInformation)
+		case let .certificate(certificateResult):
+			showScannedHealthCertificate(certificateResult)
+		case let .traceLocation(traceLocation):
+			showScannedCheckin(traceLocation)
+		}
 	}
 
+	// swiftlint:disable cyclomatic_complexity
 	private func showScannedTestResult(
 		_ testRegistrationInformation: CoronaTestRegistrationInformation
 	) {
-		switch presenter {
-		case .submissionFlow:
-			didScanCoronaTestInSubmissionFlow?(testRegistrationInformation)
-		case .onBehalfFlow:
-			let parentPresentingViewController = parentViewController?.presentingViewController
+		let recycleBinItemToRestore: RecycleBinItem?
+		switch testRegistrationInformation {
+		case .pcr(guid: _, qrCodeHash: let qrCodeHash),
+			.antigen(qrCodeInformation: _, qrCodeHash: let qrCodeHash):
+			recycleBinItemToRestore = store.recycleBinItems.first {
```

Comment:
Can we use the func `func item(for identifier: String)` from `RecycleBin` here?

Reply:
Sadly not because the `recycleBinIdentifier` for a coronaTest can't just be the `qrCodeHash` as that property is optional and the recycle bin needs to work for items that have this set to nil as well.